nodes/node.go

```diff
@@ -4,24 +4,41 @@ import (
 	"fmt"
 )
 
-// Interface for defining unit of work to be processed by event loop
+type WorkflowContext struct {
+	predecessorNodesResults []interface{}
+	nodeToResult map[InterfaceNode]interface{}
```

Comment:
Here we're using 2 data structures to maintain a single state: the predecessor results

What would happen if a DummyNode instance called SaveResult multiple times? It will replace it's entry in `nodeToResult` yet append a new entry to `predecessorNodesResults`

I think it *should* overwrite whatever old result was written before without changing the length of `predecessorNodesResults`

The way Python BRT solves this issues is like so:
- On a per-workflow basis (in Workflow or WorkflowContext), keep an array of Node objects `nodes` which can be created upon the workflow's initialization
- Keep a `nodeToResult` map which maps Node objects to their results on a per-workflow basis
- When you want to `GetResults`, iterate over `nodes` and for each node, lookup it's results from the `nodeToResult` map
- `SetResults` is simply workflowContext.nodeToResult[this] = thisResults

Reply:
This way we can maintain an "order" of predecessors Nodes and their results without weird edge cases

Reply:
I see, thanks for the explanation. I have followed this pattern as closely as possible.